src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/NoResourcePublicConstructorQuickFix.java

```diff
@@ -18,64 +18,98 @@
 import io.openliberty.tools.intellij.lsp4mp4ij.psi.core.java.corrections.proposal.ChangeCorrectionProposal;
 import org.eclipse.lsp4j.CodeAction;
 import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.WorkspaceEdit;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 /**
  * Quick fix for NoResourcePublicConstructorQuickFix that uses
  * ModifyModifiersProposal.
- * 
+ *
  * @author Shaunak Tulshibagwale
  *
  */
-public class NoResourcePublicConstructorQuickFix {
+public class NoResourcePublicConstructorQuickFix implements IJavaCodeActionParticipant {
+
+    private static final Logger LOGGER = Logger.getLogger(NoResourcePublicConstructorQuickFix.class.getName());
 
-    private final static String TITLE_MESSAGE = Messages.getMessage("MakeConstructorPublic");
+    @Override
+    public String getParticipantId() {
+        return NoResourcePublicConstructorQuickFix.class.getName();
+    }
 
     public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic) {
 
         PsiElement node = context.getCoveredNode();
         PsiMethod parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
-        
+
         if (parentMethod != null) {
             List<CodeAction> codeActions = new ArrayList<>();
 
+            codeActions.add(JDTUtils.createCodeAction(context,diagnostic,Messages.getMessage("MakeConstructorPublic"), getParticipantId()));
+            codeActions.add(JDTUtils.createCodeAction(context, diagnostic, Messages.getMessage("NoargPublicConstructor"), getParticipantId()));
+
+            return codeActions;
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+
+        final CodeAction toResolve = context.getUnresolved();
+        PsiElement node = context.getCoveredNode();
+        PsiMethod parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
+
+        if (parentMethod != null) {
+
             JavaCodeActionContext targetContext = context.copy();
             node = targetContext.getCoveredNode();
             PsiClass parentType = PsiTreeUtil.getParentOfType(node, PsiClass.class);
             parentMethod = PsiTreeUtil.getParentOfType(node, PsiMethod.class);
 
-            ChangeCorrectionProposal proposal = new ModifyModifiersProposal(TITLE_MESSAGE, targetContext.getSource().getCompilationUnit(),
-                    targetContext.getASTRoot(), parentType, 0, parentMethod.getModifierList(), Collections.singletonList("public"));
+            String message = toResolve.getTitle();
+
+            if (message.equals(Messages.getMessage("MakeConstructorPublic"))) {
+
+                ChangeCorrectionProposal proposal = new ModifyModifiersProposal(Messages.getMessage("MakeConstructorPublic"), targetContext.getSource().getCompilationUnit(),
```

Comment:
I noticed the same messages are being retrieved/computed multiple times. The value of the message was just checked above. You can pass that string directly to the constructor.